datafusion/execution/src/registry.rs

```diff
@@ -53,3 +65,51 @@ pub trait SerializerRegistry: Send + Sync {
         bytes: &[u8],
     ) -> Result<Arc<dyn UserDefinedLogicalNode>>;
 }
+
+/// A  [`FunctionRegistry`] that uses in memory [`HashMap`]s
+#[derive(Default, Debug)]
+pub struct MemoryFunctionRegistry {
```

Comment:
This is currently only used for testing, but I plan to reduce the duplication between SessionContext, and TaskContext using it